### ISSUE TITLE
Testing notes

### ISSUE THREAD

**Issue by jwzimmer** [LLK/scratch-gui#233]
@carljbowman and I were talking about some behaviours in 2.0 that can be confusing, and that we might not want to re-implement in 3.0. It's possible that some of this behaviour really does make sense as-is, but I want to at least make note of it in case others think it seems odd as well.

### Questions/ confusing behaviour (in 2.0), overview
* [Confusing project & thumbnail states](https://github.com/LLK/scratch-gui/issues/233#issuecomment-285706181)
* [When you zoom-in or move a sprite or costume so that it extends outside of the paint editor, its edges get cut off](https://github.com/LLK/scratch-gui/issues/233#issuecomment-285714778)
* Which scripting area am I in? Sometimes it's easy to unintentionally add scripts to the wrong location.
* Clearer way to interact with the numbered drums?
* Play vs. play until done
* If backdrops are analogous to costumes, not sprites, how should they be displayed?
* Should color effects transfer between backdrops & costumes? They do in 2.0, I'm just wondering about it. It might make the most sense. : )
* [Names vs. numbers - "switch costume to [costumeX]" block accepts numerical input in 2.0](https://github.com/LLK/scratch-gui/issues/233#issuecomment-289035456)
* In the paint editor (bitmap mode), it's hard to keep track of where your mouse cursor is if you're working with a light color or similar colors.

### Operating System and Browser

_Mac & Chrome_

**Comment by jwzimmer** [comment]
### Confusing project & thumbnail states

**Example projects**
* [Stamp cats - for testing](https://scratch.mit.edu/projects/147710357/)
* [Mosaic cats - for testing](https://scratch.mit.edu/projects/148052106/)
* [Foo Bar remix - cat swarm copy - for testing](https://scratch.mit.edu/projects/147720146/)

**Description**
<img width="632" alt="screen shot 2017-03-10 at 10 51 51 am" src="https://cloud.githubusercontent.com/assets/1320456/23801866/9997703c-057f-11e7-8541-d2911147745e.png">
When I stopped the project, Stamp cats - for testing, in the screenshot above, the cat must have been located somewhere off the edge of the project, since it is not visible in the screenshot.

In My Stuff, the project thumbnail displays the cat at some location, presumably where it stopped moving in one of the previous times the project was played, with no graphic effects. The stamped cats are not included in the thumbnail.
<img width="351" alt="screen shot 2017-03-10 at 10 54 34 am" src="https://cloud.githubusercontent.com/assets/1320456/23802013/f8991fd6-057f-11e7-92ce-cb835c291351.png">

When I return to this project via See Inside, the cat starts from the location shown in the thumbnail, even though I am not setting this value in the script, and it is not the location where the cat left off the last time I played and stopped the project (even if I “save now” the project).
<img width="609" alt="screen shot 2017-03-10 at 10 55 04 am" src="https://cloud.githubusercontent.com/assets/1320456/23802039/0beb1ec2-0580-11e7-8aae-fc80d09ea3b6.png">

When I view inside this project, Mosaic cats - for testing, Sprite 1 starts off screen, and the backdrop is blank. When I press the green flag, the sprite starts being stamped (on screen, near the green flag, since that’s where my mouse pointer is). When I press the stop sign, the regular sprite appears and the stamps stay visible on screen.
<img width="482" alt="screen shot 2017-03-10 at 10 55 34 am" src="https://cloud.githubusercontent.com/assets/1320456/23802051/1d6280be-0580-11e7-8a42-54afe243671e.png">

However, when I view the project in My Stuff, the thumbnail shows many of the stamps, again presumably from some previous run of the project.
<img width="448" alt="screen shot 2017-03-10 at 10 56 02 am" src="https://cloud.githubusercontent.com/assets/1320456/23802067/2e73d5c4-0580-11e7-8ea0-47291ec84344.png">

The Foo Bar remix - cat swarm copy - for testing project has the same confusing (to me) behaviour. In My Stuff, it shows a thumbnail where the cat sprite is located above and to the right of the center of the project pane, and the background is covered with stamps & clones. When I start the project, the stamps & clones are not visible, but the sprite does start off at that location (above and to right of center). I did not set this location or rotation angle of the sprite in the script itself. The state I last stop the project does not necessarily change the thumbnail in My Stuff or initial location of the sprite in the project when I next start it (after a refresh or leaving the page). (As long as I start/ stop the project without refreshing the editor or leaving the page, the state is preserved: the sprite starts/ stop at the same location and the stamps & clones are not cleared.)
<img width="356" alt="screen shot 2017-03-10 at 10 56 31 am" src="https://cloud.githubusercontent.com/assets/1320456/23802094/3f506d94-0580-11e7-9ff0-cc9e9adc2181.png">

**Comment by jwzimmer** [comment]
### When you zoom-in or move a sprite or costume so that it extends outside of the paint editor, its edges get cut off

<img width="356" alt="screen shot 2017-03-10 at 11 11 37 am" src="https://cloud.githubusercontent.com/assets/1320456/23803321/1ea76e7c-0584-11e7-831d-6fa2f6d6bf8a.png">
<img width="1141" alt="screen shot 2017-03-10 at 11 12 03 am" src="https://cloud.githubusercontent.com/assets/1320456/23803322/1ea88c76-0584-11e7-8e51-66d8348f65a4.png">
<img width="769" alt="screen shot 2017-03-10 at 11 12 22 am" src="https://cloud.githubusercontent.com/assets/1320456/23803319/1ea05e5c-0584-11e7-9c22-7c65849d8d93.png">
<img width="762" alt="screen shot 2017-03-10 at 11 12 38 am" src="https://cloud.githubusercontent.com/assets/1320456/23803318/1e9fc7a8-0584-11e7-8825-f1387085fdf1.png">
<img width="606" alt="screen shot 2017-03-10 at 11 12 56 am" src="https://cloud.githubusercontent.com/assets/1320456/23803320/1ea11fb8-0584-11e7-8ce8-7095d3656e39.png">
<img width="746" alt="screen shot 2017-03-10 at 11 13 06 am" src="https://cloud.githubusercontent.com/assets/1320456/23803316/1e995d5a-0584-11e7-8a6c-6875496e7f19.png">
<img width="440" alt="screen shot 2017-03-10 at 11 13 16 am" src="https://cloud.githubusercontent.com/assets/1320456/23803317/1e9b72e8-0584-11e7-893e-97bfee2fefcb.png">
<img width="683" alt="screen shot 2017-03-10 at 11 13 44 am" src="https://cloud.githubusercontent.com/assets/1320456/23803314/1e965326-0584-11e7-9959-ba6c88a83bc3.png">
<img width="268" alt="screen shot 2017-03-10 at 11 13 58 am" src="https://cloud.githubusercontent.com/assets/1320456/23803313/1e965f4c-0584-11e7-8a06-f139d693fcf1.png">
<img width="347" alt="screen shot 2017-03-10 at 11 14 09 am" src="https://cloud.githubusercontent.com/assets/1320456/23803312/1e95a23c-0584-11e7-84d8-e47c1c6e4966.png">
<img width="958" alt="screen shot 2017-03-10 at 11 14 23 am" src="https://cloud.githubusercontent.com/assets/1320456/23803310/1e902866-0584-11e7-9f95-8177203dfcc3.png">
<img width="1129" alt="screen shot 2017-03-10 at 11 14 55 am" src="https://cloud.githubusercontent.com/assets/1320456/23803315/1e97951a-0584-11e7-8374-c88c06632ff3.png">
<img width="1174" alt="screen shot 2017-03-10 at 11 15 13 am" src="https://cloud.githubusercontent.com/assets/1320456/23803311/1e916c80-0584-11e7-89cc-f513b956b097.png">
<img width="610" alt="screen shot 2017-03-10 at 11 15 49 am" src="https://cloud.githubusercontent.com/assets/1320456/23803308/1e8cec00-0584-11e7-98e2-22ed0557f90e.png">
<img width="1201" alt="screen shot 2017-03-10 at 11 16 00 am" src="https://cloud.githubusercontent.com/assets/1320456/23803309/1e8d88cc-0584-11e7-94a3-95f5952dcb69.png">
<img width="364" alt="screen shot 2017-03-10 at 11 16 18 am" src="https://cloud.githubusercontent.com/assets/1320456/23803305/1e87179e-0584-11e7-8fbf-d6b5e55ee1af.png">
<img width="342" alt="screen shot 2017-03-10 at 11 16 32 am" src="https://cloud.githubusercontent.com/assets/1320456/23803302/1e83227e-0584-11e7-9f5a-c46968610614.png">
<img width="214" alt="screen shot 2017-03-10 at 11 16 54 am" src="https://cloud.githubusercontent.com/assets/1320456/23803306/1e87eb2e-0584-11e7-80e4-8a4291971d6f.png">
<img width="235" alt="screen shot 2017-03-10 at 11 17 02 am" src="https://cloud.githubusercontent.com/assets/1320456/23803303/1e858604-0584-11e7-8150-cad664e318da.png">
<img width="1117" alt="screen shot 2017-03-10 at 11 14 38 am" src="https://cloud.githubusercontent.com/assets/1320456/23803307/1e882c1a-0584-11e7-92c2-20f10a6affc0.png">
<img width="1247" alt="screen shot 2017-03-10 at 11 15 31 am" src="https://cloud.githubusercontent.com/assets/1320456/23803304/1e869850-0584-11e7-95ac-97ce33451652.png">

**Comment by jwzimmer** [comment]
### Names vs. numbers - "switch costume to [costumeX]" block accepts numerical input in 2.0

I like this! It makes it easy to randomize which costume is being displayed. I wonder if it's confusing or unexpected, though, that the dropdown options include the names of the costumes, but will accept the random block / a number. The default costume names are costume1, costume2... costumeN. The number at the end also works for referring to the costume. Is this clear to users? Is there a clearer way to set this up?

<img width="129" alt="screen shot 2017-03-24 at 10 19 14 am" src="https://cloud.githubusercontent.com/assets/1320456/24298159/8a1bf266-107b-11e7-8466-de221ebc2f5d.png">
<img width="301" alt="screen shot 2017-03-24 at 10 18 57 am" src="https://cloud.githubusercontent.com/assets/1320456/24298158/8a1bae5a-107b-11e7-8e51-7eb8b7d89416.png">
<img width="298" alt="screen shot 2017-03-24 at 10 15 12 am" src="https://cloud.githubusercontent.com/assets/1320456/24298157/8a1a3b1a-107b-11e7-8546-6a98f2de0708.png">
<img width="1095" alt="screen shot 2017-03-24 at 10 14 45 am" src="https://cloud.githubusercontent.com/assets/1320456/24298156/8a1a334a-107b-11e7-9972-114ebe350f59.png">

**Comment by chrisgarrity** [comment]
Moved to LLK/observations#33

**Comment by towerofnix** [comment]
The observations repository is private. There's a lot of interesting discussion in these posts; would it be reasonable to have the repo be public? (I'm pretty sure there are settings to prevent non-collaborators from talking in a repository, if you're getting too much noise from the general Scratch community posting on observation issues, but then again, I rarely saw anybody comment on playtest issues in this public repository.)

**Comment by chrisgarrity** [comment]
@towerofnix we moved to having observations in a private repo over concerns about privacy for the young scratchers that are participating in the workshops we are having to try out Scratch 3.0 prototypes. We did realize that we'd loose comments from others in the community. We saw that as a disadvantage, not too much noise, but it is very important to us that we don't inadvertently disclose private information of young people.

The [wrapping-blocks suggestion](https://github.com/LLK/scratch-blocks/issues/400) was moved because it was tagged 'playtest', so it got caught up in the general clean-up. As you suggest, community members should continue to make suggestions in the public repos (scratch-blocks, scratch-gui etc), and we can have discussion there. I would also suggest that if we do decide to move something into observations, from now on, we leave a comment first to say why we're moving it.
